src/pod.ts

```diff
@@ -140,3 +146,29 @@ export class Pod {
     return this.cache.yamlSchema;
   }
 }
+
+export class Storage {
```

Comment:
@stevenle - do you think this level of complexity is OK? I know it's not "very" complex but I also really want to avoid introducing complexity. It's really hard to jump back into the Grow codebase now due to all of the multiple levels of caching and multiple locations you have to go to trace through the code.

For example, Steven and I were chatting a few days back about exploring/converting the file reads from `readFileSync` to `readFile` plus callbacks. It'd be harder to do that with this new abstraction introduced (IMO). I also think contributors would have a more challenging time tracing through the code versus having simple start and stop markers around the operations that need profiling.

Reply:
I think it's important we prefer async calls where they're available, for file reads/writes, url fetches, etc. Most things in ama are gonna be cpu bound and with the nature of javascript being single threaded, we should free up the cpu to do other tasks when files are being read or written.

Reply:
I think it would be good to do the `fileRead` in the pod as async. Then we could have a `fileReadSync` for the case where people don't want to use the async version.

Reply:
Yeah; we need to prototype whether `fileRead` vs `fileReadAsync` results in performance differences and make both interfaces available to callers if so. I just want to avoid the extra layers of abstraction and keep things simple here so contributors don't need to jump through code unnecessarily. The wrapping adds mental overhead to refactoring the `fileRead` now so I don't want to introduce it if we can avoid it.

So I think for the profiler code... I'd prefer if the operations in this PR were left as-is and if we just had start and stop markers for timers around the existing code. If implemented like that, it will then both be easy to add profiling to plugins, and start/stop markers don't interfere with refactoring `fileRead` to `fileReadAsync` or understanding how it works. So if/when we need to add a Storage object or change other things about how this works in the future, I recommend we can do that in a subsequent PR...

Reply:
Ok,  I will remove the Storage class and just do the `start` and `stop` around the operations.